sample_project/Assets/SampleViewer/Samples/BuildingFilter/BuildingFilter.cs

```diff
@@ -0,0 +1,502 @@
+    {
+        loadingText.gameObject.SetActive(true);
+        yield return new WaitForSeconds(time);
+
+        if (!isAdded)
+        {
+            buildingSceneLayer = arcGISMapComponent.Map.Layers.Last() as ArcGISBuildingSceneLayer;
+            if (buildingSceneLayer != null)
+            {
+                var firstLayers = buildingSceneLayer.Sublayers;
+                for (ulong i = 0; i < firstLayers.GetSize(); i++)
+                {
+                    if (firstLayers.At(i).Name == "Full Model")
+                    {
+                        firstLayers.At(i).IsVisible = true;
+                    }
+                    else if (firstLayers.At(i).Name == "Overview")
+                    {
+                        firstLayers.At(i).IsVisible = false;
+                        firstLayers.At(i).IsVisible = false;
+                    }
+                }
+            }
+        }
+
+        switch (GetLoadStatus())
+        {
+            case "Loading":
+                StartCoroutine(Delay(time, true, newLayer, solidDef, filter));
+                break;
+
+            case "Failed":
+                loadingText.gameObject.SetActive(false);
+                failedToLoadText.gameObject.SetActive(true);
+                yield return new WaitForSeconds(5);
+                failedToLoadText.gameObject.SetActive(false);
+                break;
+
+            case "Loaded":
+                loadingText.gameObject.SetActive(false);
+                AddDisciplineCategoryData();
+                contentBoxes.RemoveDisciplines();
+                contentBoxes.AddDisciplines(DisciplineCategoryData);
+                GetStatistics();
+                filter.SolidFilterDefinition = solidDef;
+                buildingSceneLayer.BuildingAttributeFilters.Add(filter);
+                buildingSceneLayer.ActiveBuildingAttributeFilter = filter;
+                var layerCenter = buildingSceneLayer.Extent.Center;
+                arcGISMapComponent.OriginPosition = layerCenter;
+                cameraLocation.Position = layerCenter;
+                phaseText.text = buildingStatistics.createdPhaseMax.ToString();
+                trashButton.onClick.Invoke();
+                GenerateWhereClause(buildingStatistics.bldgLevelMax, buildingStatistics.createdPhaseMax, true, true);
+                break;
+        }
+    }
+
+    private String GetLoadStatus()
+    {
+        if (buildingSceneLayer == null)
+        {
+            return "Failed";
+        }
+        var loadStatus = buildingSceneLayer.LoadStatus;
+
+        if (loadStatus == ArcGISLoadStatus.Loaded)
+        {
+            return "Loaded";
+        }
+        else if (loadStatus == ArcGISLoadStatus.Loading)
+        {
+            return "Loading";
+        }
+        else
+        {
+            var index = arcGISMapComponent.Map.Layers.GetSize() - 1;
```

Comment:
This whole pattern seems odd to me.

Reply:
Got rid of the entire GetLoadStatus() function.